### DISCUSSION TITLE
is cooler balance default using ICE normalization

### DISCUSSION THREAD

**Discussion by jiangshan529** [open2c/cooler#419] [Q&A] [answered]
Hi, Is cooler balance default using ICE normalization? How can I view the ICE normalized matrix and used it for downstream analysis? Thanks!

**Comment by nvictus** [comment, marked as answer]
Yes, ICE (really, just [**iterative correction**](https://www.nature.com/articles/nmeth.2148) (IC) -- the E is a separate and unrelated step) is an implementation of an operation on matrices known as **matrix balancing** in the numerical linear algebra community and [iterative proportional fitting](https://en.wikipedia.org/wiki/Iterative_proportional_fitting) in the statistics community. There's an insightful [blog post](https://liorpachter.wordpress.com/2013/11/17/imakaev_explained/) by Lior Pachter on the subject. 

The IC algorithm is essentially a variant of the [Sinkhorn-Knopp](https://en.wikipedia.org/wiki/Sinkhorn%27s_theorem#Sinkhorn%E2%80%93Knopp_algorithm) algorithm tailored to a symmetric matrix. Other algorithms, like [KR](https://strathprints.strath.ac.uk/42589/1/fastbalance.pdf), solve the exact same numerical problem but have different theoretical convergence rates and stability tradeoffs. 

Cooler balance's default normalization is matrix balancing, calculated using the IC method, applied genome-wide (rather than by chromosome in cis). 

Depending how you [query](https://cooler.readthedocs.io/en/latest/api.html#cooler.Cooler.matrix) the matrix, the normalization can be applied automatically. If you query the raw counts, you can also apply the normalization manually by multiplying each pixel `A_ij` by the corresponding balancing weights `w_i`, `w_j`. The weights are stored in the bin table as a column called `weight`.

Hope that answers your question.
